src/main/java/edu/hm/hafner/analysis/parser/PyLintParser.java

```diff
@@ -13,42 +16,59 @@
  * A parser for the PyLint compiler warnings.
  *
  * @author Sebastian Hansbauer
+ * @author Björn Pedersen
  */
 public class PyLintParser extends FastRegexpLineParser {
     private static final long serialVersionUID = 4464053085862883240L;
 
-    private static final String PYLINT_ERROR_PATTERN = "(.*):(\\d+): \\[(\\D\\d*).*\\] (.*)";
+    // the default pattern matches "--output-format=parseable" output.
+    private static final String DEFAULT_PYLINT_ERROR_PATTERN = "(?<path>.*):(?<line>\\d+): \\[(?<category>\\D\\d*)\\((?<symbol>.*)\\), .*?\\] (?<message>.*)";
 
+
+    public static final String UNKNOWN_CAT = "pylint-unknown";
     /**
      * Creates a new instance of {@link PyLintParser}.
      */
     public PyLintParser() {
-        super(PYLINT_ERROR_PATTERN);
+        super(DEFAULT_PYLINT_ERROR_PATTERN);
+    }
+
+    public PyLintParser(@CheckForNull String regExp) {
+        super(regExp);
     }
 
+
     @Override
     protected boolean isLineInteresting(final String line) {
         return line.contains("[");
     }
 
     @Override
     protected Optional<Issue> createIssue(final Matcher matcher, final IssueBuilder builder) {
-        String message = matcher.group(4);
-        String category = guessCategoryIfEmpty(matcher.group(3), message);
+        String message = matcher.group("message");
+        String category = matcher.group("symbol");
+        if (StringUtils.isEmpty(category)) {
+               category = UNKNOWN_CAT;
+        }
+        Severity priority = mapPriority(matcher.group("category"));
 
-        return builder.setFileName(matcher.group(1))
-                .setLineStart(matcher.group(2))
+        return builder.setFileName(matcher.group("path"))
+                .setLineStart(matcher.group("line"))
                 .setCategory(category)
                 .setMessage(message)
-                .setSeverity(mapPriorty(category))
+                .setSeverity(priority)
                 .buildOptional();
     }
 
-    private Severity mapPriorty(final String category) {
+    private Severity mapPriority(final String category) {
         //First letter of the Pylint classification is one of F/E/W/R/C. E/F/W are high priority.
         Severity priority;
 
         // See http://docs.pylint.org/output.html for definitions of the categories
+        if (category.isEmpty()) {
```

Comment:
Better use `StringUtils` as `category` might be `null`

Reply:
Done